test/clj/chronograph/domain/timer_test.clj

```diff
@@ -0,0 +1,480 @@
+           (db-time-span/find-all-for-timer db/datasource timer-id))
+          "all its associated time spans are deleted."))))
+
+(deftest delete-timer-isolation-by-user-test
+  (let [admin-id (:owner-id @test-context)
+        member-id (:member-id @test-context)
+        task-id (:task-id-1 @test-context)]
+    (testing "Users trying to delete each others' timers."
+      (let [timer-by-admin (timer/create! db/datasource
+                                          admin-id
+                                          task-id
+                                          "A note by an admin.")
+            timer-by-member (timer/create! db/datasource
+                                           member-id
+                                           task-id
+                                           "A note by a member.")]
+        (is (nil? (timer/delete! db/datasource
+                                 member-id
+                                 (:timers/id timer-by-admin)))
+            "A member cannot delete an admin's timer.")
+        (is (nil? (timer/delete! db/datasource
+                                 admin-id
+                                 (:timers/id timer-by-member)))
+            "An admin cannot delete a member's timer.")
+        ;; now double-check that each can delete their respective timers
+        (is (s/valid? :timers/id
+                      (timer/delete! db/datasource
+                                     admin-id
+                                     (:timers/id timer-by-admin)))
+            "An admin can delete their own timer.")
+        (is (s/valid? :timers/id
+                      (timer/delete! db/datasource
+                                     member-id
+                                     (:timers/id timer-by-member)))
+            "A member can delete their own timer.")))))
+
+
+;; ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; Update Timer's Note Tests
+;; ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+
+(deftest update-timer-note-test
+  (testing "updating a Timer's note"
+    (let [member-id (:member-id @test-context)
+          {timer-id :timers/id :as timer} (timer/create! db/datasource
+                                                         member-id
+                                                         (:task-id-1 @test-context)
+                                                         "A sample note.")
+          revised-note (str "Revised note " (rand))]
+      (is (= (dissoc (assoc timer
+                            :timers/note
+                            revised-note)
+                     :timers/updated-at)
+             (dissoc (timer/update-note! db/datasource
+                                         member-id
+                                         timer-id
+                                         revised-note)
+                     :timers/updated-at))
+          "returns a Timer containing the updated note.")
+      (is (pos? (.compareTo (:timers/updated-at
```

Comment:
If the system clock syncs after the timer was created & before the note is updated, this test could fail. Consider freezing time to make this absolutely deterministic.

Reply:
Froze time and removed the `pos?` assert.